src/ucs/vfs/fuse/vfs_fuse.c

```diff
@@ -239,7 +239,15 @@ static ucs_status_t ucs_vfs_fuse_wait_for_path(const char *path)
     /* Create inotify channel */
     ucs_vfs_fuse_context.inotify_fd = inotify_init();
     if (ucs_vfs_fuse_context.inotify_fd < 0) {
-        ucs_error("inotify_init() failed: %m");
+        if (errno == EMFILE) {
+            ucs_diag("inotify_init() failed: %m\n"
+                     "the user limit on the total number of inotify instances "
```

Comment:
errno is set to EMFILE in multiple cases: the user limit on the total number of inotify instances has been reached; the per-process limit on the number of open file descriptors has been reached; the system-wide limit on the total number of open files has been reached.
that is why I use 'may' in the diagnostic message:
`the user limit on the total number of inotify instances may has been reached`

How about this?

inotify_init failed(): %m.
please try to increase sysctl fs.inotify.max_user_instances to avoid possible inotify instances overflow.

Reply:
1. "may has" -> may have
2. maybe add function ucs_sys_check_fd_limit() to open dummy fd (dev null for example) . if it fails, print "too many open files" otherwise print "too many inotify instances, please increase sysctl fs.inotify.max_user_instances״.